packages/core/helper-plugin/src/components/GenericInput.tsx

```diff
@@ -22,13 +22,94 @@ import {
 } from '@strapi/design-system';
 import { Eye, EyeStriked } from '@strapi/icons';
 import formatISO from 'date-fns/formatISO';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import { useFieldHint } from '../hooks/useFieldHint';
+import { FieldSchema, useFieldHint } from '../hooks/useFieldHint';
 import { useFocusInputField } from '../hooks/useFocusInputField';
 import { pxToRem } from '../utils/pxToRem';
 
+import type { TranslationMessage } from '../types';
+import type { Attribute } from '@strapi/strapi';
+
+interface InputOption {
+  metadatas: {
+    intlLabel: TranslationMessage;
+    disabled: boolean;
+    hidden: boolean;
+  };
+  key: string;
+  value: string;
+}
+
+type StrapiAttribute =
+  | Attribute.String
+  | Attribute.BigInteger
+  | Attribute.Boolean
+  | Attribute.JSON
+  | Attribute.RichText
+  | Attribute.Decimal
+  | Attribute.Float
+  | Attribute.Enumeration
+  | Attribute.Integer
+  | Attribute.UID
+  | Attribute.Password
+  | Attribute.Text
+  | Attribute.Email
+  | Attribute.Date
+  | Attribute.Time
+  | Attribute.Timestamp;
+
+type InputType =
+  | 'json'
+  | 'bool'
+  | 'checkbox'
+  | 'datetime'
+  | 'date'
+  | 'number'
+  | 'email'
+  | 'timestamp'
+  | 'text'
+  | 'string'
+  | 'password'
+  | 'select'
+  | 'textarea'
+  | 'time';
+
+interface CustomInputProps extends GenericInputProps {
+  ref?: React.Ref<HTMLElement>;
+  hint?: string | React.JSX.Element | (string | React.JSX.Element)[];
+}
+
+interface GenericInputProps {
+  attribute?: StrapiAttribute;
+  autoComplete?: string;
+  customInputs?: Record<string, React.ComponentType<CustomInputProps>>;
+  description?: TranslationMessage;
+  disabled?: boolean;
+  error?: string | TranslationMessage;
+  intlLabel: TranslationMessage;
+  labelAction?: React.ReactNode;
+  name: string;
+  onChange: (
+    payload: {
+      target: {
+        name: string;
+        value: string | number | boolean | null;
+        type?: InputType;
+      };
+    },
+    shouldSetInitialValue?: boolean
+  ) => void;
+  options?: InputOption[];
+  placeholder?: TranslationMessage;
+  required?: boolean;
+  step?: number;
+  type: InputType;
+  // TODO: The value depends on the input type, too complicated to handle all cases here
+  value?: any;
```

Comment:
Is unknown possible?

Reply:
We've refined it now so it's supposed to be inferred from the attribute you pass :) it's not perfect, but i think to do this properly, we'd probably need to refactor & do it in situ of the app to get all the angles.